### PR TITLE
Adding Example for Istio Local rate limits 

### PR DESCRIPTION
**Pull request by cam634**
**Please provide a description of this PR:**
Istio's local rate limiting features with the use of envoy can be a little hard to implement at first, so why I decided to add a more in-depth example.
 
Wanted to added a small example of Istio local rate limited due to #42664, Just added an easy example some with comments to help understand how the Local rate limit filter and reduce any further issues on the topic. Can also add more detailed description to istio.io to showcase the different features to istio local rate limiting.

### CODE REVIEW

**Review by cam634 — COMMENTED** [review]
/retest-required

File: samples/ratelimit/local-rate-limit-service.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      patch:
+        operation: MERGE
+        value:
+          route:
+            rate_limits:
+              - actions:
+                  - remote_address: {}
+              - actions:
+                  - header_value_match:
+                      #descriptor_key: path_ip
```
Comment (zirain) [inline samples/ratelimit/local-rate-limit-service.yaml:48]: can you remove this line or change L83?

Comment (cam634) [reply samples/ratelimit/local-rate-limit-service.yaml]: Agreed removed line

**Review by zirain — APPROVED** [review]
cc @istio/wg-docs-maintainers-english

File: samples/ratelimit/local-rate-limit-service.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# This is a example of using istio local rate limiting while using descriptors to limit by path. 
+# It uses the base book info demo and we add rate limiting by path specifically rate limiting the product page 
+# to 10 request per minute, and the overall fdqn will be able to take 100 request in a minute. 
+# This can be applied by saving it to a file then "kubectl apply -f <filename.yaml>".
```
Comment (kfaseela) [inline samples/ratelimit/local-rate-limit-service.yaml:4]:
```suggestion
# This example shows how to use Istio local rate limiting with descriptors to limit by path. 
# This uses the base book-info demo and adds rate limiting by path, specifically rate limiting the product page 
# to 10 requests per minute, and the overall fdqn will be able to accept 100 requests per minute. 
```

**Comment by renyingxin** [comment]
Hi, the local-rate-limit-service sample gives an example for specified interface, but how to configure a rate limit rule that can limit each interface, that means for interfaces A,B,C,D I need to specify a rule that can control each interface (100 qps each etc)

**Comment by mks27** [comment]
@kfaseela @cam634 Hi, can we add multiple paths in same envoyfilter in the path based local rate limiting? -https://github.com/istio/istio/blob/master/samples/ratelimit/local-rate-limit-service.yaml

**Comment by mks27** [comment]
@kfaseela @cam634 @zirain Hi, If i am using above path based local rate-limiting then below are the observations:
1) for /productpage path , the configured limit is 10 req/min, so 10 req will be processed and for 11th request it will give 429.
and for all other paths 100 req/min is configured which is cumulative in nature. so if i already hit the  /productpage path 10 times then only 90 requests for all other paths will be handled and for more requests it will give 429. 
2)if i hit path other than /productpage then total 100 requests combining all paths coming to our pod will be handled and for more requests ,it will give 429.

but if am using the same envoyfilter at gateway level then below are the observations:
1) for /productpage path , 10 requests will be processed and for more requests , it will give 429.
2) for each other path - 100 requests will be handled and for more requests it will give 429. here it is not cumulative, it is working separately for each path.

This is the expected behaviour?

**Pull request merged**